server.js

```diff
@@ -53,26 +66,44 @@ app.post('/login_or_create_user', function (req, res) {
 
 // This is the endpoint the link in the magic link hits. It takes the token from the
 // link's query params and hits the stytch authenticate endpoint to verify the token is valid
-app.get('/authenticate', function (req, res) {
+app.get('/authenticate', async (req, res) => {
     const queryObject = url.parse(req.url,true).query;
-    client.magicLinks.authenticate(queryObject.token)
-        .then(
-            // on success render the logged in view
-            res.render('loggedIn')
-        )
-        .catch(err => {
-            // on failure, log the error then render the homepage
-            console.log(err)
-            res.render('loginOrSignUp')
-        });
+
+    const resp = await client.magicLinks.authenticate({
+        session_duration_minutes: 60,
+        token: queryObject.token,
+    })
+    if (resp.status_code !== 200) {
+        console.error('Authentication error')
+        res.status(500).send();
+        return;
+    }
+    req.session.sessionToken = resp.session_token
+    res.redirect('/')
 })
 
 // handles the logout endpoint
 app.get('/logout', function (req, res) {
+    req.session.destroy();
     res.render('loggedOut');
 })
 
 // run the server
 app.listen(port, () => {
     console.log(`Listening to requests on ${path}`);
 });
+
+// helper for getting the session authenticated user
+async function getAuthenticatedUser(req) {
+    const sessionToken = req.session.sessionToken;
+    if (!sessionToken) {
+        return;
+    }
+    const resp = await client.sessions.authenticate({session_token: sessionToken});
```

Comment:
If the authentication succeeds I think you'll want to store the refreshed token from this API call on the session, replacing whatever was there before:
```javascript
req.session.sessionToken = resp.session_token;
```

Reply:
Good catch!